src/Symfony/Component/Messenger/Worker.php

```diff
@@ -156,15 +164,15 @@ private function handleMessage(Envelope $envelope, ReceiverInterface $receiver,
                     ->with(new RedeliveryStamp($retryCount, $transportName))
                     ->withoutAll(ReceivedStamp::class);
 
-                // re-send the message
+                // re-send the message for retry
                 $this->bus->dispatch($retryEnvelope);
-                // acknowledge the previous message has received
-                $receiver->ack($envelope);
```

Comment:
I changed this to `reject` and moved it outside the `if else`. Reject is more appropriate. Technically ack and reject are the same. There is only a semantical difference. Here we are handling a failure case and whether we retry or not does not change this.

Reply:
that seems reasonable.